File: backend/src/main/java/com/github/martinhermes86/backend/model/PlantDto.java
package com.github.martinhermes86.backend.model;

import java.time.LocalDate;

public record PlantDto(
        String name,
        String species,
        String description,
        LocalDate lastWatered,
        LocalDate lastFertilized,
        LocalDate nextWatering,
        LocalDate nextFertilizing,
        String careInstructions,
        String soilRequirements,
        String locationRequirements,
        String fertilizingInstructions
) {
}

File: frontend/src/types/PlantDto.ts
export type PlantDto = {
    name: string,
    species: string,
    description: string,
    lastWatered: Date,
    lastFertilized : Date,
    nextWatering: Date,
    nextFertilizing: Date,
    careInstructions: string,
    soilRequirements: string,
    locationRequirements: string,
    fertilizingInstructions: string
}